### DISCUSSION TITLE
Conditional Custom Fields or Custom Links

### DISCUSSION THREAD

**Discussion by vstokkenes** [netbox-community/netbox#11760] [Ideas]
Greetings everyone. 

I would like to propose a new functionality that would at least help my own installation of Netbox.

I would like the ability to use conditional "Custom Fields" or "Custom Links" by using tags as condition on the Custom Field/Links. This way the Custom Field or Custom Link would only be visible for Netbox objects with the appropriate tag. 

For instance:

- You could add HSRP/VRRP specific documentation to only IP addresses tagged with the tag "HSRP-VRRP"
- You could add load balancer specific documentation to only IP addresses tagged with the dag "Citrix Netscaler" or "F5 BIG-IP LTM"
- You could tie customer specific documenation to devices tagged with "Customer name"

Would this functionality help someone else, and in that case could someone elaborate what changes that needs to be done in order to make a feature request?

Thanks alot in advance, and have a great day!

**Comment by rodvand** [comment]
You can do this today with custom links just by using jinja2 code and adding some conditionals in it. If the rendered text is blank the link won't show. Example: https://github.com/netbox-community/netbox/issues/3375

The conditional custom fields request has been brought up multiple times before and never really materialised. I think you would need to have a strong use case and a suggestion for implementation of the feature to have it considered. I would be happy to see the feature implemented, but I've managed to handle my use cases by grouping the custom fields.
